com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/clients/ServiceGetter.java

```diff
@@ -22,19 +28,26 @@ public ServiceGetter(@Qualifier("serviceInstanceGetter") AbstractServiceGetter s
         this.userProvidedServiceInstanceGetter = userProvidedServiceInstanceGetter;
     }
 
-    public Map<String, Object> getServiceInstance(CloudControllerClient client, String serviceName, String spaceId) {
-        Map<String, Object> serviceInstance = serviceInstanceGetter.getServiceInstance(client, serviceName, spaceId);
-        if (serviceInstance == null || serviceInstance.isEmpty()) {
-            serviceInstance = userProvidedServiceInstanceGetter.getServiceInstance(client, serviceName, spaceId);
-        }
-        return serviceInstance;
-    }
-
     public Map<String, Object> getServiceInstanceEntity(CloudControllerClient client, String serviceName, String spaceId) {
         Map<String, Object> serviceInstance = serviceInstanceGetter.getServiceInstanceEntity(client, serviceName, spaceId);
         if (serviceInstance == null || serviceInstance.isEmpty()) {
             serviceInstance = userProvidedServiceInstanceGetter.getServiceInstanceEntity(client, serviceName, spaceId);
         }
         return serviceInstance;
     }
+
+    public ServiceInstanceWithServiceKeys getServiceInstancesWithServiceKeys(CloudControllerClient client, CloudServiceInstance service) {
```

Comment:
getServiceInstance(S), plural? Furthermore, read below about the ServiceInstanceWithServiceKeys class